core/src/main/java/io/questdb/cairo/CairoEngine.java

```diff
@@ -200,9 +219,13 @@ public boolean releaseAllReaders() {
     }
 
     public boolean releaseAllWriters() {
-        return writerPool.releaseAll();
+        boolean released = writerPool.releaseAll();
+        if (null != backupWriterPool) {
+        	return backupWriterPool.releaseAll() || released;
```

Comment:
this should be `&&`

if writerPool releases and backup doesn't result should be false.

Perhaps a unit test should be added to ensure this behaviour

Reply:
I added a comment about this yesterday, it got lost in all the branch mess. writerPool.releaseAll() returns false if there are no writers OR if the release can't meet the deadline, but the deadline is infinite, so it will return false even iif the method succeeds (there were no writers).

Also, the return value of releaseAllWriters is never checked and its is always called when releaseAllReaders is called (by tests only). Wouldn't it be better to have a releaseAll with a void return?

Reply:
return value is used by WriterMaintenanceJob to throttle threads